### PR TITLE
add packer template

### LINKED ISSUE #51: Packer template to setup a WarpSQL Instance
Reference: https://developer.hashicorp.com/packer/docs/templates/hcl_templates

Details:

1) A packer template that allows us to build WarpSQL image for multiple platforms like Docker, AWS EC2 AMI, Azure VMI, etc.
2) The template should be flexible enough. For ex. User should be able to configure extensions users want to enable may be in ENV or Config, and only those extensions should be setup.
3) Both of the above should be done for extensions which are currently enabled.

### PR DESCRIPTION
**Pull request by PiyushRaj927**
Fixes #51 
This PR adds/improves 
1. a Packer template that enables building images with installed extensions using a parametric approach. By default, all supported extensions are installed. To install specific extensions, you can provide the extensions variable when running the command:
```bash
packer build -var extensions='pg_repack,hll' -only warpsql.docker.alpine war
```
2. GitHub workflow to test each extension individually as well as all of them within an image. By default, the test script `test/smoke_test/smoke_test.sh` will execute all the scripts in the `extensions_test` directory.  You can specify which tests to run by providing a comma-separated list of extensions as an argument to the script. Here's an example of how to run specific tests:
```bash
./test/smoke_test/smoke_test.sh "hll,pg_reapck"
```

3. updated `README.md` file with instructions on how to use the Packer template.

### CODE REVIEW

**Comment by PiyushRaj927** [comment]
cc: @singhalkarun

**Comment by PiyushRaj927** [comment]
Hey @singhalkarun, could you please review the changes?

**Comment by singhalkarun** [comment]
@PiyushRaj927 Can you please check why the actions are running infinitely? Also can you brief in the description of PR itself what the action does? Generally, going forward, please brief all the changes done in PR Description.

Running Infinitely
Running

**Comment by PiyushRaj927** [comment]
@singhalkarun I have made updates to the PR description and included a workflow to test each extension individually. Please review these changes.

**Comment by PiyushRaj927** [comment]
Regarding the issue with the infinitely running action, it appears that there was an outage on GitHub (you can check the details here: [GitHub Status](https://www.githubstatus.com/incidents/tcp7119jvq7y)). This outage might have caused a bug to arise while running the specific action. I'm not entirely sure how to fix it.

File: .github/workflows/smoke-test-packer.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
+name: Smoke Test packer template
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    # branches:
```
Comment (jayanth-kumar-morem) [inline .github/workflows/smoke-test-packer.yml:10]: Revert 😅

File: README.md [hunk 2]
```diff
@@ -44,6 +44,46 @@ By utilizing WarpSQL, you can benefit from the following:
...
+2. Build the images:
+    ```shell 
+      packer build warpsql.pkr.hcl
+    ``` 
+To build only the Alpine image, you can use the `-only` option:
+
+  ```shell
+  packer build -only=warpsql.docker.alpine warpsql.pkr.hcl
+  ```
+  By default, all supported extensions are installed. If you want to install specific extensions, you can provide the `extensions` variable:
```
Comment (jayanth-kumar-morem) [inline README.md:76]: Please add a line mentioning the list of extensions they can use at the moment.

File: test/smoke_test/extentions_test/test_timescaledb.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -ex 
+
+psql <<EOF
```
Comment (jayanth-kumar-morem) [inline test/smoke_test/extentions_test/test_timescaledb.sh:4]: `psql <<EOF` and `psql -c "<>"`

Are these both the same ??

**Review by jayanth-kumar-morem — COMMENTED** [review]
Nice work @PiyushRaj927

File: test/smoke_test/extentions_test/test_timescaledb.sh (hunk 3 above)
Comment (PiyushRaj927) [reply test/smoke_test/extentions_test/test_timescaledb.sh:4]: Yes! https://www.postgresql.org/docs/current/app-psql.html

File: README.md (hunk 2 above)
Comment (PiyushRaj927) [reply README.md:76]: Sure

**Review by jayanth-kumar-morem — APPROVED** [review]
+2 LGTM, Please squash the commits @PiyushRaj927

**Review by singhalkarun — APPROVED** [review]
GTG @PiyushRaj927

**Comment by singhalkarun** [comment]
@PiyushRaj927 Can you raise a new PR with squashed commit, I merged by mistake, I will merge new PR

**Pull request merged**